src/main/java/core/basesyntax/dataservices/DataProcessImpl.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax.dataservices;
+
+import core.basesyntax.FruitTransaction;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class DataProcessImpl implements DataProcess {
+    @Override
+    public Map<String, Integer> process(List<FruitTransaction> transactions) {
+        Map<String, Integer> balanceMap = new HashMap<>();
+        for (FruitTransaction transaction : transactions) {
+            balanceMap.putIfAbsent(transaction.getFruit(), 0);
+
+            switch (transaction.getOperation()) {
+                case BALANCE -> balanceMap.put(transaction.getFruit(), transaction.getQuantity());
+                case SUPPLY, RETURN -> balanceMap.put(transaction.getFruit(),
+                        balanceMap.get(transaction.getFruit()) + transaction.getQuantity());
+                case PURCHASE -> balanceMap.put(transaction.getFruit(),
+                        balanceMap.get(transaction.getFruit()) - transaction.getQuantity());
```

Comment:
Before updating the balance for a PURCHASE operation, ensure that the result balance will not be negative. If it is, throw a RuntimeException.